src/java.base/windows/classes/sun/nio/fs/WindowsPath.java

```diff
@@ -831,12 +831,53 @@ long openForReadAttributeAccess(boolean followLinks)
         int flags = FILE_FLAG_BACKUP_SEMANTICS;
         if (!followLinks)
             flags |= FILE_FLAG_OPEN_REPARSE_POINT;
+        try {
+            return openFileForReadAttributeAccess(flags);
+        } catch (WindowsException e) {
+            if (!followLinks || e.lastError() != ERROR_CANT_ACCESS_FILE)
+                throw e;
+            // Object could be a Unix domain socket
+            try {
```

Comment:
Thanks for the updates, I think the fallback checking is correct everywhere now.
What would you think about inverting the above so that it only checks for a socket file when followLinks && e.lastError() == ERROR_CANT_ACCESS_FILE. That would make it more consistent with the new code in WIndowsFileSystemProvider.

Reply:
Okay. No problem. The next revision also includes an updated test where I found some socket files weren't being cleaned up at the end of the test.